README.adoc

```diff
@@ -8,42 +8,430 @@ The time for the full course is around three to four days.
+subject. It may cover up to 4 topics.
+
+Attendees may not become productive in all areas, but get a good feeling
+for them.
+
+This course is perfect for engineering management, that will most likely
+not end up using Rust daily, but needs broad knowledge to assess Rust as
+a technology. It is also frequently given in academic settings.
+
+5 days
+^^^^^^
+
+The longer versions covers similar subjects as above, but in more
+detail. This allows e.g. the construction of a larger piece of software
+in the training group collaboratively.
+
+These courses are recommended for people that are intended to be
+multipliers in your organisation.
+
+After 5 days, attendees will have solid knowledge in a topical domain
+and full productivity.
+
+2 days
+^^^^^^
+
+A condensed version of the three days course. Detailed topic deep dives
+will be reduced and some self-guided exercises reduced to ``code along''
+sessions.
+
+2,4 and 8 hours topic tasters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Taster sessions that favor experiences over solid knowledge and get
+people excited.
+
+Perfect for conferences and internal all-hands to expand peoples
+horizons.
+
+Advanced courses
+~~~~~~~~~~~~~~~~
+
+Advanced courses deepen a Rust subject and are indended for groups that
+already visited a course before. They quickly refreshen core knowledge
+and then add a topical deep dive on topic. Advanced courses also differ
+from base courses in that they include a number facilitated discussions
+for attendees to share their experience.
+
+Advanced courses are tailored like base courses to the need of the
+group. For example, an basic course for for asynchronous programming
+explains how concurrent execution in Rust works, the advanced one may
+include writing your own execution engine.
+
+Topics
+------
+
+Ferrous Systems offers Rust courses for a wide variety of subjects. The
+following is a non-exhaustive list of subjects.
+
+Every subject with be taught with hands-on exercises.
+
+Core topics
+~~~~~~~~~~~
+
+This is the core component of the course. In this course, attendees will
+learn:
+
+* Ownership
+* Borrowing
+* Lifetimes
+* Working with memory and data structures
+* Control flow
+* Structuring of applications
+* Using and configuring cargo and rustup
+* Introducing Rust into existing products
+* General Rust patterns
+* Error Handling
+* Using Generic APIs
+* Using Rusts guarantees for security gains
+* Testing and Debugging
+* Documentation tooling
+* Basics of concurrency safety
+* Overview of important documentation
+* Overview the standard library, especially core interface like
+collections, input/output and networking
+* Overview of common libraries
+* ``Refactoring towards Speed'': a core technique to safely derive a
+fast program from a working program
+
+Exercises will be tailored to the chosen special subjects.
+
+Advanced Generic Programming in Rust
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using libraries with generics in Rust is taught in the core course. When
```

Comment:
I suggest replacing this paragraph with:

> Although the use of generics is taught in the core course, this section instead focuses on how, and when, to use them effectively. For developers of widely used libraries (internal or public), this material is particularly important.

Reply:
Applied manually.